desc/compute/_metric.py

```diff
@@ -2088,3 +2088,565 @@ def _cvdrift0(params, transforms, profiles, data, **kwargs):
+    # TODO: This can be computed more efficiently without building radial
+    #       derivatives of the stream functions by taking vartheta derivative
+    #       of formula used to compute sqrt(g)_PEST.
+    data["(sqrt(g)_PEST_v)|PEST"] = (
+        dot(data["(e_rho_v)|PEST"], cross(data["e_theta_PEST"], data["e_phi|r,v"]))
+        + dot(
+            data["e_rho|v,p"], cross(data["(e_theta_PEST_v)|PEST"], data["e_phi|r,v"])
+        )
+        + dot(data["e_rho|v,p"], cross(data["e_theta_PEST"], data["(e_phi_v)|PEST"]))
+    )
+    return data
+
+
+# TODO: Generalize for a general phi before #568
+@register_compute_fun(
+    name="(sqrt(g)_PEST_p)|PEST",
+    label="\\partial_{\\phi}|_{\\rho, \\vartheta} \\sqrt{g}_PEST",
+    units="m^{3}",
+    units_long="cubic meters",
+    description="Jacobian determinant of PEST coordinate system"
+    + " derivative w.r.t cylindrical toroidal angle",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "e_rho|v,p",
+        "e_theta_PEST",
+        "e_phi|r,v",
+        "(e_theta_PEST_p)|PEST",
+        "(e_rho_p)|PEST",
+        "(e_phi_p)|PEST",
+    ],
+)
+def _sqrtg_PEST_phi_PEST(params, transforms, profiles, data, **kwargs):
+    # TODO: This can be computed more efficiently without building radial
+    #       derivatives of the stream functions by taking phi derivative
+    #       of formula used to compute sqrt(g)_PEST.
+    data["(sqrt(g)_PEST_p)|PEST"] = (
+        dot(data["(e_rho_p)|PEST"], cross(data["e_theta_PEST"], data["e_phi|r,v"]))
+        + dot(
+            data["e_rho|v,p"], cross(data["(e_theta_PEST_p)|PEST"], data["e_phi|r,v"])
+        )
+        + dot(data["e_rho|v,p"], cross(data["e_theta_PEST"], data["(e_phi_p)|PEST"]))
+    )
+    return data
+
+
+################################################################################
+#############--------------FINITE-n STABILITY METRICS--------------#############
+################################################################################
+
+
+@register_compute_fun(
+    name="(B*grad) grad(rho)",
+    label="\\nabla(\\nabla(\\rho))",
+    units="T m^{-2}",
+    units_long="Tesla over square meters",
+    description="Gradient of contravariant radial basis vector(grad rho)"
+    "along the magnetic field scaled by the magnetic field strength",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["e^rho_t", "e^rho_z", "B^theta", "B^zeta"],
+)
+def _B_dot_grad_grad_rho(params, transforms, profiles, data, **kwargs):
+    data["(B*grad) grad(rho)"] = (
+        data["B^theta"][:, jnp.newaxis] * data["e^rho_t"]
+        + data["B^zeta"][:, jnp.newaxis] * data["e^rho_z"]
+    )
+    return data
+
+
+@register_compute_fun(
+    name="finite-n instability drive",
+    label="(\\mathbf{J} \\times (\\nabla \\rho))/{(g^{\\rho \\rho})}^2"
+    + " \\mathbf{B} \\cdot \\cdot \\mathbf{\\nabla} (\\mathbf{\\nabla} \\rho)",
```

Comment:
multiple cdot

Reply:
I'll updated it after looking for other label and description-related typos.